### PR TITLE
Timer Widget

### PR DESCRIPTION
**Pull request by gsmithun4**
Related to https://github.com/ToolJet/ToolJet/issues/1632

### CODE REVIEW

File: frontend/src/Editor/Components/Timer.jsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    intervalId && clearInterval(intervalId);
+    setTime(getTimeObj());
+    fireEvent('onReset');
+    setState('initial');
+    setExposedVariable('value', time);
+  };
+
+  const onStart = () => {
+    setIntervalId(
+      setInterval(() => {
```
Comment (akshaysasidrn) [inline frontend/src/Editor/Components/Timer.jsx:51]: Why are we adding and subtracting with 15 on ms?

**Review by akshaysasidrn — CHANGES_REQUESTED** [review]
Please reset timer value based on count down/up and also do check linked events not being triggered.

File: frontend/src/Editor/Components/Timer.jsx (hunk 1 above)
Comment (gsmithun4) [reply frontend/src/Editor/Components/Timer.jsx:58]: `setInterval` interval is set to 15 ms

**Comment by gsmithun4** [comment]
> Please reset timer value based on count down/up and also do check linked events not being triggered.

Done

File: frontend/src/Editor/Components/Timer.jsx (hunk 1 above)
Comment (akshaysasidrn) [reply frontend/src/Editor/Components/Timer.jsx:58]: Sorry, why that particular value of 15ms ? 🤔

Comment (akshaysasidrn) [reply frontend/src/Editor/Components/Timer.jsx:58]: I suppose it is to make it easier to see the updation?

**Comment by akshaysasidrn** [comment]
Change LGTM 👍🏾

**Review by akshaysasidrn — APPROVED** [review]

**Pull request merged**